OOT! x86/of: Don't log error on the missing IO-APIC unconditionally

romank-msft opened the pull request:
The x86 OF code reports the missing IO-APIC wihtout any checks whether the IO-APIC presence is required or not. That is due to that part of the code being dovetailed to the Intel MID platform.

Check if the IO-APIC must be available. Log the error only when the IO-APIC is expected to be present.

See: https://github.com/microsoft/OHCL-Linux-Kernel/issues/37

benhillis commented:
This looks reasonable to me but I'd like to let the LSG folks weigh in here.

saurabh-sengar commented:
Since we are taking this upstream anyway and will take back the patch after review. Can we keep things simple in our code ? ie changing pr_err to pr_dbg ?

benhillis commented:
That's a good suggestion, I think either way would accomplish our goal.

romank-msft commented:
Reduced to a one-liner.

romank-msft commented:
> Since we are taking this upstream anyway and will take back the patch after review. Can we keep things simple in our code ? ie changing pr_err to pr_dbg ?

`pr_debug` LOL. Will push an update to fix pr_dbg

saurabh-sengar commented:
Looks good. If you could have a single commit in PR that will be great.
If it can be done while merging thats OK too.

romank-msft committed:
OOT! x86/of: Downgrade the error on the missing IO-APIC unconditionally

That code is dovetailed to Intel MIDs, and the error message might be
bogus for other systems. For OpenHCL, it certainly is.

Downgrade the log message. A proper fix is in review in LKML.

Signed-off-by: Roman Kisel <romank@linux.microsoft.com>

romank-msft commented:
GH can squash, I can, too :D. Pushed
